### ISSUE TITLE
[TextField] Screen reader does not announce the error message

### ISSUE THREAD

**Issue by lucas-stanford** [microsoft/fluentui#18076]
### Environment Information
OS Build: Version 20H2(19042.867)
Browser: Version 89.0.4389.90 (Official Build) (64-bit)
Screen Reader:  Narrator, also JAWS and NVDA.

### Describe the issue:
Screen reader users may not be able to figure out the message and may not be able to catch about what mistake has been done if screen reader does not read the message which appears on the screen.

### Please provide a reproduction of the issue in a codepen:
https://fabricweb.z5.web.core.windows.net/oufr/7.170.0/#/examples/textfield

#### Actual behavior:
 Screen reader does not announce the error message appearing on the screen for "With error message" TextField.

#### Expected behavior:
 Screen reader should announce the error message appearing on the screen for "With error message" TextField.

### Documentation describing expected behavior
https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html

----
Bug id: 9564665

**Comment by lucas-stanford** [comment]
dup of https://github.com/microsoft/fluentui/issues/13309
